lib/appsignal/oban.ex

```diff
@@ -160,25 +158,27 @@ defmodule Appsignal.Oban do
   end
 
   def oban_insert_job_start(_event, _measurements, metadata, _config) do
-    span = @tracer.create_span("oban", @tracer.current_span)
+    current_span = @tracer.current_span()
 
-    @span.set_attribute(span, "appsignal:category", "insert_job.oban")
+    if current_span do
+      span = @tracer.create_span("oban", @tracer.current_span())
 
-    case metadata[:changeset] do
-      %{changes: %{worker: worker}} ->
-        @span.set_name(span, "Insert job (#{worker})")
+      @span.set_attribute(span, "appsignal:category", "insert_job.oban")
 
-      _ ->
-        # The changeset containing the worker appears in the metadata for
-        # this event in every version where this event is emitted (from
-        # v2.11.0 until v2.13.6, latest at the time of writing) but it's
-        # undocumented. To be safe, account for it not being present.
-        @span.set_name(span, "Insert job")
-    end
-  end
+      case metadata[:changeset] do
+        %{changes: %{worker: worker}} ->
+          @span.set_name(span, "Insert job (#{worker})")
 
-  def oban_insert_job_stop(_event, _measurements, _metadata, _config) do
-    @tracer.close_span(@tracer.current_span())
+        _ ->
+          # The changeset containing the worker appears in the metadata for
+          # this event in every version where this event is emitted (from
+          # v2.11.0 until v2.13.6, latest at the time of writing) but it's
+          # undocumented. To be safe, account for it not being present.
+          @span.set_name(span, "Insert job")
+      end
+
+      @tracer.close_span(@tracer.current_span())
```

Comment:
You want to close the "insert_job" span immediately here? Would we lose any metadata like how long the event took?

Also, would this be faster?

```suggestion
      @tracer.close_span(span)
```

Reply:
This patch changes the implementation of the insert_job start event. Previously, that event would _always_ create a span. Because of that, both the stop and exception events could assume a span was created, and they could just close whatever the current span was. By changing the implementation to where the start event doesn't always create a span, we can no longer make that assumption.

A solution to this problem is closing the span immediately after creating it. That would (in theory) create 0 us event, as it no longer records the duration of the insertion. It also stops handling exceptions.

Reply:
Also, yes, not finding the current span again would be faster. Good find!